WordPressKitTests/DomainsServiceRemoteRESTTests.swift

```diff
@@ -238,4 +243,37 @@ class DomainsServiceRemoteRESTTests: RemoteTestCase, RESTTestable {
 
         waitForExpectations(timeout: timeout, handler: nil)
     }
+
+    // MARK: - Get All Domains Tests
+
+    func testGetAllDomainsSucceeds() {
+        let expect = expectation(description: "Get All Domains Succeeds")
+
+        stub { request -> Bool in
+            let url = request.url
+            let matched = url?.path == self.allDomainsEndpoint && url?.query == "no_wpcom=true&resolve_status=true&locale=en"
```

Comment:
I assume the check on the query here is to ensure that the parameters object is read and used correctly to build the request? 

If so, two notes. First, it's worth noting that we have no guarantees in how Foundation will decide to build the query string. I could end up being different, e.g. with `locale=en` as the first parameter. OHHTTPStubs offers a condition checker for query parameters that is robust against changes in the query string order:

```swift
public func containsQueryParams(_ params: [String:String?]) -> HTTPStubsTestBlock {
```

There is also a similar method for the path, 

```swift
public func isPath(_ path: String) -> HTTPStubsTestBlock {
```

Second, and more of a test style and readability thing, so partly subject to personal opinion. The query is defined up here and the parameters down there. It's not clear that they are related to each other. If they the params were defined before the stub and maybe the check read their value (`containsQueryParams(["no_wpcom": parameters.noWPCOM, ...`) the relationship would be clearer.

Reply:
Resolved in 9d0a555a91d0db7a158f8bc613176a8be085180c